src/analytics/baseTracker/baseTracker.ts

```diff
@@ -63,4 +67,15 @@ export class BaseTracker {
       ? document.referrer?.toLowerCase()
       : "undefined";
   }
+
+  getDomain(url: string): string {
+    const newUrl = new URL(url);
+    return `${newUrl.protocol}//${newUrl.host}`;
+  }
+
+  getDomainPath(url: string, part: number): string {
+    const newUrl = new URL(url);
+    const path = newUrl.pathname.split(/(?<=^(?:.{500})+)(?!$)/);
```

Comment:
This regex might cause us some problems down the line as it's using a positive lookbehind which isn't supported in IE11. The service manual doesn't list IE11 as a browser that services need to support but it's been flagged in one of the channels (https://gds.slack.com/archives/C0396SWLUPN/p1700137652483349?thread_ts=1700136567.797569&cid=C0396SWLUPN) that we're a little different as we need to take into account a small number of users that might be using IE11. With that in mind, `newUrl.pathname.match(/.{1,500}/g)` might be better suited?

Reply:
I've checked the regex I provided and it outputs the substrings into an array as well
<img width="625" alt="image" src="https://github.com/govuk-one-login/di-fec-ga4/assets/148477198/b980832e-f67a-408c-8134-c3fb6a5cc5d7">

Reply:
I simply replace use of a regexp by just a substring. Doing well the work without the potential hassle of a regexp